### ISSUE TITLE
Segregated witness results in significantly different VSize calculation

### ISSUE THREAD

**Issue by dcousens** [bitcoinjs/coinselect#4]

**Comment by dcousens** [comment]
Options:

* `.witness` field for each `input`, or
* `.prevOutScript`/`.prevOutType` for each `input`

The latter is probably better...

**Comment by runn1ng** [comment]
Hey, I will look at this soon, I am currently in need for coinselect to work on segwit :)

**Comment by runn1ng** [comment]
Which script types would like to support?

I could take the types from here (except for NONSTANDARD)

https://github.com/bitcoinjs/bitcoinjs-lib/blob/170328341bdcf2a22d6ee6df1df587ea4e83c8a9/src/templates/index.js#L11

**Comment by dcousens** [comment]
`.script.length` is used now for estimation [if provided] - however,  it assumes P2PKH.

**Comment by dcousens** [comment]
If you're implementation can pre-sign all inputs,  but with fake/blank signatures [for security],  then `script` and an added `witness` field would be more than enough to complete this.

**Comment by runn1ng** [comment]
Hm, but pre-signing all inputs with empty/fake privkeys will complicate things and needlessly slow down computation, when the inputs types usually have the same lengths anyway for the given type

**Comment by dcousens** [comment]
> Hm, but pre-signing all inputs with empty/fake privkeys will complicate things and needlessly slow down computation

Not necessarily,  it should be no different then the approach we'd be taking here (pattern match,  speculate byte length impact via a fake script/witness.

**Comment by dcousens** [comment]
Resolved partially by https://github.com/bitcoinjs/coinselect/pull/22#discussion_r133176608 - but - the virtual size calculation isn't accounted for yet?

**Comment by blmalone** [comment]
Is this issue being worked on? :)

**Comment by mahnunchik** [comment]
Any news?

**Comment by karelbilek** [comment]
I don’t want to talk for current maintainers, but neither me or dcousens is
in bitcoin business anymore

**Comment by mahnunchik** [comment]
What's happened?

**Comment by junderw** [comment]
I am here, but this repository is not a priority for me.

PRs and more importantly PR reviews are greatly appreciated and needed.

**Comment by ChrisCho-H** [comment]
I made PR https://github.com/bitcoinjs/coinselect/pull/82 regarding this issue.
